test/ipynb/mpl/test_circuit_matplotlib_drawer.py

```diff
@@ -138,6 +149,165 @@ def test_no_barriers_false(self):
 
         self.circuit_drawer(circuit, filename='no_barriers.png', plot_barriers=False)
 
+    def test_fold_minus1(self):
+        """Test to see that fold=-1 is no folding"""
+        qr = QuantumRegister(2, 'q')
+        cr = ClassicalRegister(1, 'c')
+        circuit = QuantumCircuit(qr, cr)
+        for _ in range(3):
+            circuit.h(0)
+            circuit.x(0)
+
+        self.circuit_drawer(circuit, fold=-1, filename='fold_minus1.png')
+
+    def test_fold_4(self):
+        """Test to see that fold=4 is folding"""
+        qr = QuantumRegister(2, 'q')
+        cr = ClassicalRegister(1, 'c')
+        circuit = QuantumCircuit(qr, cr)
+        for _ in range(3):
+            circuit.h(0)
+            circuit.x(0)
+
+        self.circuit_drawer(circuit, fold=4, filename='fold_4.png')
+
+    def test_big_gates(self):
+        """Test large gates with params"""
+        qr = QuantumRegister(6, 'q')
+        circuit = QuantumCircuit(qr)
+        A = [[6, 5, 3], [5, 4, 5], [3, 5, 1]]
+        circuit.append(IQP(A), [0, 1, 2])
+
+        desired_vector = [
+            1 / math.sqrt(16) * complex(0, 1),
+            1 / math.sqrt(8) * complex(1, 0),
+            1 / math.sqrt(16) * complex(1, 1),
+            0,
+            0,
+            1 / math.sqrt(8) * complex(1, 2),
+            1 / math.sqrt(16) * complex(1, 0),
+            0]
+
+        circuit.initialize(desired_vector, [qr[3], qr[4], qr[5]])
+        circuit.unitary([[1, 0], [0, 1]], [qr[0]])
+        matrix = np.zeros((4, 4))
+        theta = Parameter('theta')
+        circuit.append(HamiltonianGate(matrix, theta), [qr[1], qr[2]])
+        circuit = circuit.bind_parameters({theta: 1})
+        circuit.isometry(np.eye(4, 4), [i for i in range(3, 5)], [])
+
+        self.circuit_drawer(circuit, filename='big_gates.png')
+
+    def test_ctrl_x(self):
+        """Test different control-x gates"""
+        qr = QuantumRegister(5, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.x(0)
+        circuit.cx(0, 1)
+        circuit.ccx(0, 1, 2)
+        circuit.append(XGate().control(3, ctrl_state='010'), [qr[2], qr[3], qr[0], qr[1]])
+        circuit.append(MCXGate(num_ctrl_qubits=3, ctrl_state='101'), [qr[0], qr[1], qr[2], qr[4]])
+
+        self.circuit_drawer(circuit, filename='ctrl_x.png')
```

Comment:
`control-x` is not very common name for `cx`. I would suggest `cnot`
```suggestion
    def test_cnot(self):
        """Test different cnot gates"""
        qr = QuantumRegister(5, 'q')
        circuit = QuantumCircuit(qr)
        circuit.x(0)
        circuit.cx(0, 1)
        circuit.ccx(0, 1, 2)
        circuit.append(XGate().control(3, ctrl_state='010'), [qr[2], qr[3], qr[0], qr[1]])
        circuit.append(MCXGate(num_ctrl_qubits=3, ctrl_state='101'), [qr[0], qr[1], qr[2], qr[4]])

        self.circuit_drawer(circuit, filename='cnot.png')
```

Reply:
Either/or. I always assumed a cx was a cnot, a ccx was a Toffoli, and all the others were multi-controlled x's.

Reply:
Changed to cnot in 750d066b9